packages/blocks/src/store/reducer.js

```diff
@@ -372,18 +372,22 @@ export function collections( state = {}, action ) {
 }
 
 export function blockBindingsSources( state = {}, action ) {
-	if ( action.type === 'REGISTER_BLOCK_BINDINGS_SOURCE' ) {
-		return {
-			...state,
-			[ action.sourceName ]: {
-				label: action.sourceLabel,
-				getValue: action.getValue,
-				setValue: action.setValue,
-				setValues: action.setValues,
-				getPlaceholder: action.getPlaceholder,
-				canUserEditValue: action.canUserEditValue || ( () => false ),
-			},
-		};
+	switch ( action.type ) {
+		case 'ADD_BLOCK_BINDINGS_SOURCE':
+			return {
+				...state,
+				[ action.name ]: {
+					label: action.label,
+					getValue: action.getValue,
+					setValue: action.setValue,
+					setValues: action.setValues,
+					getPlaceholder: action.getPlaceholder,
+					canUserEditValue:
+						action.canUserEditValue || ( () => false ),
```

Comment:
What's the reasoning for providing a default handler only in this place? `getValue`, `setValue`, `setValues` ang `getPlaceholder` are all also optional so they will be undefined. Should the default handler for `canUserEditValue` be moved to the selector instead?

Reply:
For sources that aren't registered or don't provide that property, we want to lock the editing when a paragraph (or other block) is connected to it. 

I didn't think about moving it to the selector, I can give that a try. I guess it would require changing both `getAllBlockBindingsSources` and `getBlockBindingsSource`: [link](https://github.com/WordPress/gutenberg/blob/9e73c1158bb8eb79d6415743a7519fad2d2ffef1/packages/blocks/src/store/private-selectors.js#L197-L211).

Reply:
I created a follow-up pull request for this as well: [link](https://github.com/WordPress/gutenberg/pull/63628). I didn't move the logic to the selector, but I think I prefer that approach better. Let me know what you think 🙂